_posts/2021-06-09-ejb-over-http.adoc

```diff
@@ -0,0 +1,118 @@
+----
+
+== Configure the HTTP Transport
+
+First we must configure the http-authentication-factory for the server to authenticate over HTTP.
+Here we set the mechanism to BASIC (but could also be set to FORM), point to the fsSD security-domain, and the fsRealm filesystem-realm, and we call the newly created http-authentication-factory, example-http-auth
+
+[source]
+----
+/subsystem=elytron/http-authentication-factory=example-http-auth:add(http-server-mechanism-factory=global,security-domain=fsSD,mechanism-configurations=[{mechanism-name=BASIC,mechanism-realm-configurations=[{realm-name=fsRealm}]}])
+----
+
+Then in order to operate over HTTP, we must disable the legacy security-realm. This is done by undefining it from the http-invoker
+
+[source]
+----
+/subsystem=undertow/server=default-server/host=default-host/setting=http-invoker:undefine-attribute(name=security-realm)
+----
+
+And finally we can enable the http-invoker with the http-authentication-factory defined earlier (example-http-auth) that allows remote HTTP invocation for the EJB
+
+[source]
+----
+/subsystem=undertow/server=default-server/host=default-host/setting=http-invoker:write-attribute(name=http-authentication-factory, value=example-http-auth)
+----
+
+== Configure the Credential Store
+
+To create a credential store, we run the following command, which creates it in the path specified after the --location flag, and secures it with the password specified after the --password flag (in this case it is “StorePassword”)
+
+[source]
+----
+WILDFLY_HOME/bin/elytron-tool.sh credential-store --create --location "/PATH/TO/mycredstore.cs" --password StorePassword
+----
+
+We then create an alias to be able to access this credential store with a different username and password, in this case the username being “quickstartUser”, and the password being “quickstartPwd1!”
+
+[source]
+----
+WILDFLY_HOME/bin/elytron-tool.sh credential-store --location "/PATH/TO/mycredstore.cs" --password StorePassword --add quickstartUser --secret quickstartPwd1!
+----
+
+== Setting up the configuration - wildfly-config.xml
+[source, xml]
+----
+<credential-stores>
+    <credential-store name="mycredstore">
+        <attributes>
+            <attribute name="keyStoreType" value="JCEKS"/>
+            <attribute name="location" value="/PATH/TO/mycredstore.cs"></attribute>
+        </attributes>
+        <protection-parameter-credentials>
+            <clear-password password="StorePassword"/>
+        </protection-parameter-credentials>
+    </credential-store>
+</credential-stores>
+----
+Here we can see the definition of the credential store.
+Make sure to change the path from "/PATH/TO/mycredstore.cs", to the same path as when configuring the credential store. Also ensure that the path here is the absolute path, starting from `/`
+
+== Summary
+This blog post has given an overview on how to create an EJB over HTTP, authenticate it with a FileSystem realm, and use a credential-store to invoke the EJB.
```

Comment:
"how to create" -> "how to invoke"

"authenticate it with a FileSystem realm" -> "make use of a filesystem realm"

"use a credential-store to invoke the EJB" -> "use a credential-store to obtain the password that should be used to invoke the EJB"

Reply:
Fixed